webhook-service/src/main/scala/io/renku/webhookservice/CommitSyncRequestSender.scala

```diff
@@ -18,70 +18,34 @@
 import io.renku.webhookservice.model.CommitSyncRequest
-import org.http4s.Status.{Accepted, BadGateway, GatewayTimeout, ServiceUnavailable}
-import org.http4s.{Status, Uri}
 import org.typelevel.log4cats.Logger
 
-import scala.concurrent.duration._
-
 private trait CommitSyncRequestSender[F[_]] {
-  def sendCommitSyncRequest(commitSyncRequest: CommitSyncRequest): F[Unit]
+  def sendCommitSyncRequest(commitSyncRequest: CommitSyncRequest, processName: String): F[Unit]
 }
 
-private class CommitSyncRequestSenderImpl[F[_]: Async: Temporal: Logger](
-    eventLogUrl:            EventLogUrl,
-    retryDelay:             FiniteDuration,
-    retryInterval:          FiniteDuration = RestClient.SleepAfterConnectionIssue,
-    maxRetries:             Int Refined NonNegative = RestClient.MaxRetriesAfterConnectionTimeout,
-    requestTimeoutOverride: Option[Duration] = None
-) extends RestClient[F, CommitSyncRequestSender[F]](
-      Throttler.noThrottling,
-      retryInterval = retryInterval,
-      maxRetries = maxRetries,
-      requestTimeoutOverride = requestTimeoutOverride
-    )
-    with CommitSyncRequestSender[F] {
+private class CommitSyncRequestSenderImpl[F[_]: MonadThrow: Logger](eventSender: EventSender[F])
+    extends CommitSyncRequestSender[F] {
 
+  import eventSender._
   import io.circe.Encoder
   import io.circe.literal._
   import io.circe.syntax._
-  import org.http4s.Method.POST
-  import org.http4s.{Request, Response}
-
-  def sendCommitSyncRequest(syncRequest: CommitSyncRequest): F[Unit] = for {
-    uri           <- validateUri(s"$eventLogUrl/events")
-    sendingResult <- send(prepareRequest(uri, syncRequest))(mapResponse) recoverWith retryDelivery(syncRequest)
-    _             <- logInfo(syncRequest)
-  } yield sendingResult
-
-  private def prepareRequest(uri: Uri, commitSyncRequest: CommitSyncRequest) =
-    request(POST, uri).withMultipartBuilder
-      .addPart("event", commitSyncRequest.asJson)
-      .build()
-
-  private def retryDelivery(commitSyncRequest: CommitSyncRequest): PartialFunction[Throwable, F[Unit]] = {
-    case UnexpectedResponseException(ServiceUnavailable | GatewayTimeout | BadGateway, message) =>
-      waitAndRetry(Eval.always(sendCommitSyncRequest(commitSyncRequest)), message)
-    case exception @ (_: ConnectivityException | _: ClientException) =>
-      waitAndRetry(Eval.always(sendCommitSyncRequest(commitSyncRequest)), exception.getMessage)
-  }
 
-  private def waitAndRetry(retry: Eval[F[Unit]], errorMessage: String) = for {
-    _      <- Logger[F].error(s"Sending commit sync request event failed - retrying in $retryDelay - $errorMessage")
-    _      <- Temporal[F] sleep retryDelay
-    result <- retry.value
-  } yield result
+  def sendCommitSyncRequest(syncRequest: CommitSyncRequest, processName: String): F[Unit] =
+    sendEvent(
+      EventRequestContent.NoPayload(syncRequest.asJson),
+      EventSender.EventContext(CategoryName("COMMIT_SYNC_REQUEST"),
```

Comment:
should we store COMMIT_SYNC_REQUEST in a shared variable somewhere?

Reply:
Hm, that might be good and bad as that's the only usage in this service. All the other usages are in EL and it won't be nice to make them a hard link between these two. Anyway, we'd need to think more about how to eventually do it in other places too. For now, it somehow follows the common pattern of not sharing the model classes between different components so they are not tightly coupled.